### PR TITLE
Update logins iOS code to create carthage builds and fix tests

### PR DESCRIPTION
**Pull request by lougeniaC64**
This PR fixes/removes the iOS logins wrapper and tests to allow for iOS artifact builds for the `logins-per-field-encryption` branch. 

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - Note:
    - For changes that need extra cross-platform testing, consider adding `[ci full]` to the PR title.
    - If this pull request includes a breaking change, consider [cutting a new release](https://github.com/mozilla/application-services/blob/main/docs/howtos/cut-a-new-release.md) after merging.
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry in [CHANGES_UNRELEASED.md](../CHANGES_UNRELEASED.md) or an explanation of why it does not need one
  - Any breaking changes to Swift or Kotlin binding APIs are noted explicitly
- [ ] **Dependencies**: This PR follows our [dependency management guidelines](https://github.com/mozilla/application-services/blob/main/docs/dependency-management.md)
  - Any new dependencies are accompanied by a summary of the due dilligence applied in selecting them.

### CODE REVIEW

File: megazords/ios/MozillaAppServicesTests/LoginsTests.swift
Comment (skhamis) [inline megazords/ios/MozillaAppServicesTests/LoginsTests.swift:9]: I won't block on this, but do we have a task we can add in here to ensure we don't forget to get this in?

**Review by skhamis — APPROVED** [review]
Thanks for getting this in, Looks good to me!

**Comment by skhamis** [comment]
btw, that failing test should be fixed on the latest of `logins-per-field-encryption`

File: megazords/ios/MozillaAppServicesTests/LoginsTests.swift
Comment (lougeniaC64) [reply megazords/ios/MozillaAppServicesTests/LoginsTests.swift:9]: You should, lol! This was actually a note to myself. I'm going to add the tests before merging.

**Pull request merged**
